### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by dukaev**

### CODE REVIEW

**Comment by dukaev** [comment]
@DamirSvrtan I think **config_file** and **scannable_files** from **file_traverser.rb** should be private

File: lib/fasterer/file_traverser.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ class FileTraverser
 
     attr_reader :config
     attr_reader :parse_error_paths
-    attr_accessor :offenses_total_count
+    attr_reader :offenses_total_count
```
Comment (DamirSvrtan) [inline lib/fasterer/file_traverser.rb]: if you remove the writer, how does line 64 work? `self.offenses_total_count += analyzer.errors.count` Is it not under test?

Comment (dukaev) [reply lib/fasterer/file_traverser.rb]: @DamirSvrtan 	
You are right. it will create an error. I added a new commit.
Can you review?
Thank you.

**Comment by DamirSvrtan** [comment]
Hi @dukaev, thank you for your contribution!

Is there a spec that verifies this behavior?

As well, can you please squash the commits, I'm not a proponent of lazy git messages such as 'fix'.

**Pull request closed without merging**